schema-resolver/src/main/java/io/apicurio/registry/resolver/DefaultSchemaResolver.java

```diff
@@ -267,35 +250,16 @@ private SchemaLookupResult<S> handleResolveSchemaByContent(ParsedSchema<S> parse
         String rawSchemaString = IoUtil.toString(parsedSchema.getRawSchema());
 
         return schemaCache.getByContent(rawSchemaString, contentKey -> {
-
             logger.info(String.format("Retrieving schema content using string: %s", rawSchemaString));
 
-            InputStream is = new ByteArrayInputStream(contentKey.getBytes(StandardCharsets.UTF_8));
-            String at = schemaParser.artifactType();
-            String ct = toContentType(at);
-            VersionSearchResults results = client.search().versions().post(is, ct, config -> {
-                config.queryParameters.groupId = artifactReference.getGroupId() == null ? "default"
-                    : artifactReference.getGroupId();
-                config.queryParameters.artifactId = artifactReference.getArtifactId();
-                config.queryParameters.canonical = true;
-                config.queryParameters.artifactType = at;
-                config.queryParameters.orderby = VersionSortBy.GlobalId;
-                config.queryParameters.order = SortOrder.Desc;
-            });
-
-            if (results.getCount() == 0) {
-                is = new ByteArrayInputStream(contentKey.getBytes(StandardCharsets.UTF_8));
-                results = client.search().versions().post(is, ct, config -> {
-                    config.queryParameters.groupId = artifactReference.getGroupId() == null ? "default"
-                        : artifactReference.getGroupId();
-                    config.queryParameters.artifactId = artifactReference.getArtifactId();
-                    config.queryParameters.canonical = false;
-                    config.queryParameters.artifactType = at;
-                    config.queryParameters.orderby = VersionSortBy.GlobalId;
-                    config.queryParameters.order = SortOrder.Desc;
-                });
-
-                if (results.getCount() == 0) {
+            String artifactType = schemaParser.artifactType();
+
+            List<RegistryVersionCoordinates> versions = this.clientFacade.searchVersionsByContent(rawSchemaString,
+                    artifactType, artifactReference, true);
+            if (versions.isEmpty()) {
+                versions = this.clientFacade.searchVersionsByContent(rawSchemaString, artifactType, artifactReference, false);
+
+                if (versions.isEmpty()) {
```

Comment:
this may  be the opportunity to include changes proposed in https://github.com/Apicurio/apicurio-registry/pull/6320
And ensure the version is not disabled before continuing ?